lib/model/folder_sendrecv.go

```diff
@@ -1547,7 +1544,7 @@ func (f *sendReceiveFolder) pullBlock(state pullBlockState, snap *db.Snapshot, o
 	}
 
 	var lastError error
-	candidates := f.model.availabilityInSnapshot(f.FolderConfiguration, snap, state.file, state.block)
+	candidates := f.model.availabilityInSnapshot(f.FolderConfiguration, snap, state.file)
```

Comment:
Here we still need candidates from both the snapshot or downloads. Either you call both new methods here, or likely cleaner/simpler you create another internal function e.g. lower-case `availability` which does that, and then also use that function in `Availability`.

Reply:
Still applies - you need to call `m.availability(...` here.